server/server.go

```diff
@@ -134,6 +136,18 @@ func (s *Server) StartEtcd(apiHandler http.Handler) error {
 		return errors.Trace(err)
 	}
 
+	// Check cluster ID
+	urlmap, err := types.NewURLsMap(s.cfg.InitialCluster)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(urlmap) != 1 {
+		otherPeerURLs := etcdutil.ExtractURLsExcept(urlmap, s.cfg.Name)
```

Comment:
Can we just check all peer urls and wrap the whole logic in a function?
This `StartEtcd` function does a lot of things, we want to keep it short and clean.